### PR TITLE
add AVCaptureDeviceTypeExternal support for iPad OS 17 uvccamra

### PR DESCRIPTION
**Pull request by mtdxc**

### CODE REVIEW

File: ios/RCTWebRTC/WebRTCModule+RTCMediaStream.m [hunk 1]
```diff
@@ -263,7 +263,7 @@ - (RTCVideoTrack *)createScreenCaptureVideoTrack {
 #else
     NSMutableArray *devices = [NSMutableArray array];
     AVCaptureDeviceDiscoverySession *videoevicesSession =
-        [AVCaptureDeviceDiscoverySession discoverySessionWithDeviceTypes:@[ AVCaptureDeviceTypeBuiltInWideAngleCamera, AVCaptureDeviceTypeBuiltInUltraWideCamera, AVCaptureDeviceTypeBuiltInTelephotoCamera, AVCaptureDeviceTypeBuiltInDualCamera, AVCaptureDeviceTypeBuiltInDualWideCamera, AVCaptureDeviceTypeBuiltInTripleCamera]
+        [AVCaptureDeviceDiscoverySession discoverySessionWithDeviceTypes:@[ AVCaptureDeviceTypeBuiltInWideAngleCamera, AVCaptureDeviceTypeBuiltInUltraWideCamera, AVCaptureDeviceTypeBuiltInTelephotoCamera, AVCaptureDeviceTypeBuiltInDualCamera, AVCaptureDeviceTypeBuiltInDualWideCamera, AVCaptureDeviceTypeBuiltInTripleCamera, AVCaptureDeviceTypeExternal]
```
Comment (saghul) [inline ios/RCTWebRTC/WebRTCModule+RTCMediaStream.m:266]: We support down to iOS 13, you need to only use the symbol on iOS 17 or it won't work with older versions.

File: ios/RCTWebRTC/WebRTCModule+RTCMediaStream.m [hunk 2]
```diff
@@ -280,7 +280,7 @@ - (RTCVideoTrack *)createScreenCaptureVideoTrack {
         [devices addObject:@{
             @"facing" : position,
             @"deviceId" : device.uniqueID,
-            @"groupId" : @"",
+            @"groupId" : device.deviceType,
```
Comment (saghul) [inline ios/RCTWebRTC/WebRTCModule+RTCMediaStream.m:283]: I don't think the device type matches well the group concept, please leave it out.

File: examples/GumTestApp/App.js [hunk 3]
```diff
@@ -24,7 +24,17 @@ const App = () => {
     console.log('start');
     if (!stream) {
       try {
-        const s = await mediaDevices.getUserMedia({ video: true });
+        let opt = {video: true};
```
Comment (saghul) [inline examples/GumTestApp/App.js:27]: Can we do a device selection combo box instead?

**Review by saghul — CHANGES_REQUESTED** [review]

**Comment by mtdxc** [comment]
I have update the code, please review again

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Thanks!

**Pull request merged**